src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration/GetChildKeysTests.cs

```diff
@@ -0,0 +1,111 @@
+            for (int i = 0; i < 1000; i++)
+            {
+                emptyKeys.Add(new string(' ', i), string.Empty);
+            }
+
+            var inputKeys = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
+                inputKeys.Add(i.ToString(), string.Empty);
+            }
+
+            var splittingKeys = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
+                splittingKeys.Add("a:" + i.ToString(), string.Empty);
+            }
+
+            var keysWithCommonPaths = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
+                keysWithCommonPaths.Add("a:b:c" + i.ToString(), string.Empty);
+            }
+
+            _chainedConfigEmpty = new ChainedConfigurationProvider(new ChainedConfigurationSource
+            {
+                Configuration = new ConfigurationBuilder()
+                    .Add(new MemoryConfigurationSource { InitialData = emptyKeys })
+                    .Build(),
+                ShouldDisposeConfiguration = false,
+            });
+
+            _chainedConfig = new ChainedConfigurationProvider(new ChainedConfigurationSource
+            {
+                Configuration = new ConfigurationBuilder()
+                    .Add(new MemoryConfigurationSource { InitialData = inputKeys })
+                    .Build(),
+                ShouldDisposeConfiguration = false,
+            });
+
+            _chainedConfigWithSplitting = new ChainedConfigurationProvider(new ChainedConfigurationSource
+            {
+                Configuration = new ConfigurationBuilder()
+                    .Add(new MemoryConfigurationSource { InitialData = splittingKeys })
+                    .Build(),
+                ShouldDisposeConfiguration = false,
+            });
+
+            _chainedConfigWithCommonPaths = new ChainedConfigurationProvider(new ChainedConfigurationSource
+            {
+                Configuration = new ConfigurationBuilder()
+                    .Add(new MemoryConfigurationSource { InitialData = keysWithCommonPaths })
+                    .Build(),
+                ShouldDisposeConfiguration = false,
+            });
+        }
+
+        [GlobalCleanup(Targets = new[] {
+            nameof(AddChainedConfigurationNoDelimiter),
+            nameof(AddChainedConfigurationEmpty),
+            nameof(AddChainedConfigurationWithSplitting),
+            nameof(AddChainedConfigurationWithCommonPaths)
+        })]
+        public void CleanupBasic()
+        {
+            _chainedConfig.Dispose();
+            _chainedConfigEmpty.Dispose();
+            _chainedConfigWithSplitting.Dispose();
+            _chainedConfigWithCommonPaths.Dispose();
+        }
+
+        [Benchmark]
+        public void AddChainedConfigurationNoDelimiter() => _chainedConfig.GetChildKeys(new string[0], null);
```

Comment:
if we use `new string[0]` here, the allocation will be included in memory reported by BDN.

```suggestion
        public void AddChainedConfigurationNoDelimiter() => _chainedConfig.GetChildKeys(Array.Empty<string>(), null);
```